benchmark/fs/bench-rmdirSync.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+const common = require('../common');
+const fs = require('fs');
+const tmpdir = require('../../test/common/tmpdir');
+tmpdir.refresh();
+
+const bench = common.createBenchmark(main, {
+  type: ['existing', 'non-existing'],
+  n: [1e3],
+});
+
+function main({ n, type }) {
+  switch (type) {
+    case 'existing': {
+      for (let i = 0; i < n; i++) {
+        const p = tmpdir.resolve(`rmdirsync-bench-dir-${i}`);
+        fs.mkdirSync(p, { recursive: true });
+      }
+
+      bench.start();
+      for (let i = 0; i < n; i++) {
+        const p = tmpdir.resolve(`rmdirsync-bench-dir-${i}`);
+
+        fs.rmdirSync(p, {
+          recursive: true,
+          maxRetries: 3,
```

Comment:
This will succeed in first try. You can openSync a file and not close it until the last retry.

Reply:
I tried this, but rmSync still succeeds. The underlying file gets unlinked, but the descriptor still points to a valid inode, so no retries are triggered. From the current set of retry codes (EBUSY, EMFILE, ENFILE, ENOTEMPTY, EPERM), EBUSY, ENOTEMPTY and EPERM seem like the best candidates to try and trigger.

I first attempted triggering ENOTEMPTY, but noticed it wasn't possible as the implementation of rimraf.js performs readDir and recurses before rimrafUnlinkSync is called. The way to trigger the new path is to call rmSync on a file instead of dir using recursive option(isDirectory() check is performed before entering the cpp part).

For EBUSY or EPERM, I'm not sure what would be a good way to benchmark  them with via rmSync, maybe you have idea?